receiver/hostmetricsreceiver/hostmetrics_receiver_test.go

```diff
@@ -133,3 +139,35 @@ func assertMetricDataAndGetMetricsSlice(t *testing.T, metrics pdata.Metrics) pda
 	assert.Equal(t, 1, ilms.Len())
 	return ilms.At(0).Metrics()
 }
+
+func benchmarkScrapeMetrics(b *testing.B, cfg *Config) {
+	sink := &exportertest.SinkMetricsExporter{}
+
+	receiver, _ := newHostMetricsReceiver(context.Background(), zap.NewNop(), cfg, factories, sink)
+	receiver.initializeScrapers(context.Background(), componenttest.NewNopHost())
+
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		receiver.scrapeMetrics(context.Background())
+	}
+
+	if len(sink.AllMetrics()) == 0 {
+		b.Fail()
+	}
+}
+
+func Benchmark_ScrapeMetricsAll(b *testing.B) {
+	cfg := &Config{
+		CollectionInterval: 100 * time.Millisecond,
+		Scrapers: map[string]internal.Config{
+			cpuscraper.TypeStr:        &cpuscraper.Config{ReportPerCPU: true},
```

Comment:
Use default config in benchmarks since that will be the most common

Reply:
Added a benchmark for defaults and a benchmark for all